src/nrfx/hal/nrf_egu.c

```diff
@@ -0,0 +1,136 @@
+/** Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Note that the function prototypes are taken from the NRFx HAL
+ */
+
+#include "hal/nrf_egu.h"
+#include "bs_tracing.h"
+#include "NRF_EGU.h"
+#include <kernel.h>
+#include <irq_ctrl.h>
+
+uint32_t nrf_egu_channel_count(NRF_EGU_Type const *p_reg)
+{
+	return sizeof(p_reg->EVENTS_TRIGGERED) / sizeof(p_reg->EVENTS_TRIGGERED[0]);
+}
+
+void nrf_egu_task_trigger(NRF_EGU_Type *p_reg, nrf_egu_task_t egu_task)
+{
+	*((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)egu_task)) = 0x1UL;
```

Comment:
Minor: magic number

Reply:
IMHO 1 and 0 in this context is acceptable